services/user/Identity/service/src/tests/queries.rs

```diff
@@ -1,227 +1,84 @@
-            subject: AccountNumber::from("bob"),
-            value: 75,
-        },
-        PartialAttestation {
-            attester: AccountNumber::from("alice"),
-            subject: AccountNumber::from("carol"),
-            value: 77,
-        },
-        PartialAttestation {
-            attester: AccountNumber::from("carol"),
-            subject: AccountNumber::from("bob"),
-            value: 76,
-        },
+    let svc = init_svc::<Identity, Identity>(&chain);
+    chain.new_account("carol".into())?;
+
+    svc.from("carol").attest("bob".into(), 76).get()?;
+    svc.from("alice").attest("carol".into(), 76).get()?;
+    svc.from("alice").attest("carol".into(), 77).get()?;
+    svc.from("alice").attest("bob".into(), 75).get()?;
+
+    let all = svc.query::<Vec<Attestation>>("allAttestations", &[]);
+    let all_expected = vec![
+        PartialAttestation::new("alice", "bob", 75),
+        PartialAttestation::new("alice", "carol", 77),
+        PartialAttestation::new("carol", "bob", 76),
+    ];
+    assert_equal(&all_expected, &all);
+
+    let all_stats = svc.query::<Vec<AttestationStats>>("allAttestationStats", &[]);
+    let all_stats_expected = vec![
+        PartialAttestationStats::new("bob", 2, 1),
+        PartialAttestationStats::new("carol", 1, 1),
     ];
+    assert_equal(&all_stats_expected, &all_stats);
 
-    let response = query_attestations(
-        &chain,
-        "allAttestations",
-        get_gql_query_attestations_no_args(String::from("allAttestations")),
-    );
-    assert!(are_equal_vecs_of_attestations(&exp_results, &response));
-
-    let exp_result2 = exp_results
-        .clone()
-        .into_iter()
-        .filter(|pa| pa.subject == AccountNumber::from("bob"))
-        .collect::<Vec<PartialAttestation>>();
-
-    let response = query_attestations(
-        &chain,
-        "attestationsBySubject",
-        get_gql_query_attestations_one_arg("attestationsBySubject", "subject", "bob"),
-    );
-    assert!(are_equal_vecs_of_attestations(&exp_result2, &response));
-
-    let exp_result2 = exp_results
-        .into_iter()
-        .filter(|pa| pa.attester == AccountNumber::from("alice"))
-        .collect::<Vec<PartialAttestation>>();
-    let response = query_attestations(
-        &chain,
-        "attestationsByAttester",
-        get_gql_query_attestations_one_arg("attestationsByAttester", "attester", "alice"),
-    );
-    assert!(are_equal_vecs_of_attestations(&exp_result2, &response));
-
-    let exp_results = vec![
-        PartialAttestationStats {
-            subject: AccountNumber::from("bob"),
-            uniqueAttesters: 2,
-            numHighConfAttestations: 1,
-        },
-        PartialAttestationStats {
-            subject: AccountNumber::from("carol"),
-            uniqueAttesters: 1,
-            numHighConfAttestations: 1,
-        },
+    let by_subject = svc.query::<Vec<Attestation>>("attestationsBySubject", &[("subject", "bob")]);
```

Comment:
When the parameters are all literals, I don't think you gain anything compared to `r#"attestationsBySubjective(subject: "bob")"#`.

Reply:
That's true, and then, what, I guess just extract anything before the first "`(`" to get the `query_name` for indexing into the response (`res["data"][query_name]`)

Reply:
Done